tests/blockchain/eth/entities/test_actors.py

```diff
@@ -2,98 +2,95 @@
-    _origin, ursula, *everybody_else = chain.provider.w3.eth.accounts
-    miner = Miner(miner_agent=mock_miner_agent, address=ursula)
-    return miner
+class TestMiner:
 
+    @pytest.fixture(scope='class')
+    def miner(self, chain, mock_token_agent, mock_miner_agent):
+        mock_token_agent.token_airdrop(amount=100000 * mock_token_agent.M)
+        _origin, ursula, *everybody_else = chain.interface.w3.eth.accounts
+        miner = Miner(miner_agent=mock_miner_agent, address=ursula)
+        return miner
 
-def test_miner_locking_tokens(chain, miner, mock_miner_agent):
+    def test_miner_locking_tokens(self, chain, miner, mock_miner_agent):
 
-    assert mock_miner_agent.min_allowed_locked < miner.token_balance(), "Insufficient miner balance"
+        assert mock_miner_agent.min_allowed_locked < miner.token_balance(), "Insufficient miner balance"
 
-    miner.stake(amount=mock_miner_agent.min_allowed_locked,    # Lock the minimum amount of tokens
-                lock_periods=mock_miner_agent.min_locked_periods)   # ... for the fewest number of periods
+        miner.stake(amount=mock_miner_agent.min_allowed_locked,         # Lock the minimum amount of tokens
+                    lock_periods=mock_miner_agent.min_locked_periods)   # ... for the fewest number of periods
 
-    # Verify that the escrow is "approved" to receive tokens
-    assert mock_miner_agent.token_agent.contract.functions.allowance(miner.address, mock_miner_agent.contract_address).call() == 0
+        # Verify that the escrow is "approved" to receive tokens
+        assert mock_miner_agent.token_agent.contract.functions.allowance(miner.address, mock_miner_agent.contract_address).call() == 0
 
-    # Staking starts after one period
-    assert mock_miner_agent.contract.functions.getLockedTokens(miner.address).call() == 0
+        # Staking starts after one period
+        assert mock_miner_agent.contract.functions.getLockedTokens(miner.address).call() == 0
 
-    # Wait for it...
-    chain.time_travel(periods=1)
-    assert mock_miner_agent.contract.functions.getLockedTokens(miner.address).call() == mock_miner_agent.min_allowed_locked
-
-
-def test_miner_collects_staking_reward_tokens(chain, miner, mock_token_agent, mock_miner_agent, mock_policy_agent):
-
-    # Capture the current token balance of the miner
-    initial_balance = miner.token_balance()
-    assert mock_token_agent.get_balance(miner.address) == miner.token_balance()
+        # Wait for it...
+        chain.time_travel(periods=1)
+        assert mock_miner_agent.contract.functions.getLockedTokens(miner.address).call() == mock_miner_agent.min_allowed_locked
 
-    # Have other address lock tokens
-    _origin, *everybody_else = chain.provider.w3.eth.accounts
-    mock_miner_agent.spawn_random_miners(addresses=everybody_else)
+    @pytest.mark.slow()
+    def test_miner_collects_staking_reward_tokens(self, chain, miner, mock_token_agent, mock_miner_agent):
```

Comment:
`mock_policy_agent` in parameters is used for the policy manager initialization (`setPolicyManager`)
`mint` do not work without configured policy manager in the `MinersEscrow`